File: dataset/output/GEMINI_gemini-1.5-pro-001/Task186_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
 public:
  ListNode* mergeKLists(vector<ListNode*>& lists) {
    if (lists.empty()) return nullptr;
    auto comp = [](ListNode* a, ListNode* b) { return a->val > b->val; };
    priority_queue<ListNode*, vector<ListNode*>, decltype(comp)> pq(comp);
    for (ListNode* head : lists) {
      if (head) pq.push(head);
    }
    ListNode* dummy = new ListNode(-1);
    ListNode* cur = dummy;
    while (!pq.empty()) {
      ListNode* top = pq.top();
      pq.pop();
      cur->next = top;
      cur = cur->next;
      if (top->next) pq.push(top->next);
    }
    return dummy->next;
  }
};

File: dataset/output/GEMINI_gemini-1.5-pro-001/task186/Task186_GEMINI_gemini_1_5_pro_001.java
/**
 * Definition for singly-linked list.
 * public class Task186_GEMINI_gemini_1_5_pro_001 {
 *     int val;
 *     Task186_GEMINI_gemini_1_5_pro_001 next;
 *     Task186_GEMINI_gemini_1_5_pro_001() {}
 *     Task186_GEMINI_gemini_1_5_pro_001(int val) { this.val = val; }
 *     Task186_GEMINI_gemini_1_5_pro_001(int val, Task186_GEMINI_gemini_1_5_pro_001 next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public Task186_GEMINI_gemini_1_5_pro_001 mergeKLists(ListNode[] lists) {
        if (lists == null || lists.length == 0) {
            return null;
        }
        PriorityQueue<ListNode> minHeap = new PriorityQueue<>((a, b) -> a.val - b.val);
        for (ListNode node : lists) {
            if (node != null) {
                minHeap.add(node);
            }
        }
        Task186_GEMINI_gemini_1_5_pro_001 dummy = new Task186_GEMINI_gemini_1_5_pro_001(-1);
        Task186_GEMINI_gemini_1_5_pro_001 cur = dummy;
        while (!minHeap.isEmpty()) {
            Task186_GEMINI_gemini_1_5_pro_001 node = minHeap.poll();
            cur.next = node;
            cur = cur.next;
            if (node.next != null) {
                minHeap.add(node.next);
            }
        }
        return dummy.next;
    }
}